app/lib/features/events/widgets/events_item.dart

```diff
@@ -35,7 +48,10 @@ class EventItem extends ConsumerWidget {
               ),
             ),
             const SizedBox(width: 10),
-            _buildRsvpStatus(context, ref),
+            Visibility(
+              visible: isShowRsvp,
+              child: _buildRsvpStatus(context, ref),
```

Comment:
Why not use ```Consumer``` in this widget method instead of making ```EventItem``` widget a consumer state? I don't ```ref``` usage in ```EventItem``` itself.